proof/encoding.go

```diff
@@ -492,14 +499,108 @@ func GroupKeyRevealDecoder(r io.Reader, val any, buf *[8]byte, l uint64) error {
 		if err != nil {
 			return err
 		}
+
+		// Compute remaining bytes. This calculation will not underflow
+		// because we have already verified that the length is at least
+		// the size of the compressed public key.
 		remaining := l - btcec.PubKeyBytesLenCompressed
-		var tapscriptRoot []byte
-		err = tlv.DVarBytes(r, &tapscriptRoot, buf, remaining)
-		if err != nil {
-			return err
+
+		// Attempt to read the tapscript root bytes if they are present.
+		var tapscriptRootBytes []byte
+		if remaining >= 32 {
```

Comment:
In retrospect, I think we should've gone with a nest TLV here, so one value of the key, and another of the tapscript root. Then we wouldn't need this special casing to try and detect what was encoded.

Reply:
We also have an option of using a new TLV type here, making them mutually exclusive. 

Do we know how an old client will react if the old reveal is just _missing_ all together (no type specified).